build.gradle

```diff
@@ -26,10 +26,14 @@ repositories {
     mavenCentral()
 }
 
+// This extracts the first number before any dot or dash
+def extractMajorVersion(String version) {
```

Comment:
Please inline this function into the single caller. It does not seem worthwhile to split this up into two.